RecoEgamma/EgammaPhotonProducers/src/ConversionTrackCandidateProducer.cc

```diff
@@ -279,6 +282,8 @@ void ConversionTrackCandidateProducer::produce(edm::Event& theEvent, const edm::
     validEndcapSCHandle = false;
   }
 
+  thresholds = &theEventSetup.getData(ecalPFRechitThresholdsToken_);
```

Comment:
Why does this need to be a pointer and can not be a constant reference?

Reply:
it was just easier, as later on we can check if it is nullptr or not, and depending on that we do certain things. 
We can deliberately pass a nullptr in cases where we do not want to apply the PF recHit thresholds at a later stage.